src/schema/v2/me/artistTypeUserInterests.ts

```diff
@@ -0,0 +1,53 @@
+import { GraphQLFieldConfig, GraphQLInt } from "graphql"
+import { connectionFromArraySlice } from "graphql-relay"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { pageable } from "relay-cursor-paging"
+import { ResolverContext } from "types/graphql"
+import { createPageCursors } from "../fields/pagination"
+import {
+  userInterestCategoryEnum,
+  UserInterestConnection,
+} from "../userInterests"
+
+export const artistTypeUserInterests: GraphQLFieldConfig<
+  void,
+  ResolverContext
+> = {
+  type: UserInterestConnection,
+  args: pageable({
+    category: {
```

Comment:
Rather than add the `category` support for the `/artists` endpoint in https://github.com/artsy/gravity/pull/16430 , maybe it should be added to the more general endpoint: https://github.com/artsy/gravity/blob/8bcff15733d54588a464b74d5abb74286bdc9a94/app/api/v1/me_user_interests_endpoint.rb#L11

(sidenote I don't know why we need that /artists endpoint at all, the more general one will return that data with an `interestType: 'Artist'`)

...And also add the `interestType` argument here (Eigen will use it with `interestType: 'Artist'`).

That way, we have a nice `interestConnection` under `me` that can be effectively used for any purpose.

The Eigen query would probably be something like:

``` graphql
me {
  interestsConnection(first: X, interestType: 'Artist') { 
    edges {
      # edge fields
      node {
        ... on Artist {
          # artist selections
        }
      }
    }
  }
}
```
```